rmgpy/solver/base.pyx

```diff
@@ -892,48 +916,46 @@ cdef class ReactionSystem(DASx):
                                 bimolecularThreshold[i,j] = True
 
                 # Interrupt simulation if that flux exceeds the characteristic rate times a tolerance
-            if (not ignoreOverallFluxCriterion) and (maxSpeciesRate > toleranceMoveToCore * charRate and not invalidObject):
+            if not (maxSpecies in invalidObjects) and (not ignoreOverallFluxCriterion) and (maxSpeciesRate > toleranceMoveToCore * charRate) and len(invalidObjects) < maxNumObjsPerIter:
```

Comment:
I think `maxSpecies not in invalidObjects` is more readable, but the two methods are completely equivalent so it's up to you whether or not to change it.